asg/src/scope.rs

```diff
@@ -178,16 +182,15 @@ impl<'a> Scope<'a> {
             leo_ast::Type::Array(sub_type, dimensions) => {
                 let item = Box::new(self.resolve_ast_type(&*sub_type, span)?);
 
-                use leo_ast::ArrayDimensions::*;
-                match dimensions {
-                    Unspecified => Type::ArrayWithoutSize(item),
-                    Number(num) => Type::Array(
+                match dimensions.deref() {
+                    [Dimension::Unspecified] => Type::ArrayWithoutSize(item),
+                    [Dimension::Number(num)] => Type::Array(
                         item,
                         num.to_string()
                             .parse::<u32>()
                             .map_err(|_| AsgError::parse_index_error(span))?,
                     ),
-                    Multi(_) => unimplemented!("multi array type should not exist at asg level"),
+                    _ => unimplemented!("multi array type should not exist at asg level"),
```

Comment:
Should this be `unreachable`?
```suggestion
                    _ => unreachable!("multi array type should not exist at asg level"),
```